src/lib/oper_data.c

```diff
@@ -182,37 +179,279 @@ int get_list_keys(const struct lysc_node_list *list, json_object *json_array_obj
 
                 if (!*key_values || !*values_lengths) {
                     fprintf(stderr, "Failed to allocate memory\n");
-                    return EXIT_FAILURE;
+                    ret = EXIT_FAILURE;
+                    goto cleanup;
                 }
 
                 // Store the new key
                 (*key_values)[key_count] = strdup(json_object_get_string(temp_value));
                 if (!(*key_values)[key_count]) {
                     fprintf(stderr, "Failed to duplicate list key value\n");
-                    return EXIT_FAILURE;
+                    ret = EXIT_FAILURE;
+                    goto cleanup;
                 }
                 (*values_lengths)[key_count] = strlen(json_object_get_string(temp_value));
                 key_count++;
             } else {
                 // key value not found in json data.
-                return EXIT_FAILURE;
+                ret = EXIT_FAILURE;
```

Comment:
we are not freeing `key_values` and `values_lengths` here.

Reply:
Its the caller responsibility to free them